### PR TITLE
Add option to enable/disable runtime plugin loading

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,6 +68,7 @@ option(ENABLE_BENCHMARKS "Build benchmarks" Off)
 option(RAJA_DEPRECATED_TESTS "Test deprecated features" Off)
 option(RAJA_ENABLE_BOUNDS_CHECK "Enable bounds checking in RAJA::Views/Layouts" Off)
 option(RAJA_TEST_EXHAUSTIVE "Build RAJA exhaustive tests" Off)
+option(ENABLE_RUNTIME_PLUGINS "Enable support for loading plugins at runtime" Off)
```
Comment (mdavis36) [inline CMakeLists.txt:71]: I'm happy leaving these off by default, but we probably want to make sure we run some of the travis tests with these on

Comment (rhornung67) [reply CMakeLists.txt:71]: Agreed. @davidbeckingsale we have an issue to clean up our CMake variables by prepending them with 'RAJA_' to make it clear which ones we own. Will you please change the name of this one following that?

Comment (davidbeckingsale) [reply CMakeLists.txt]: Should be all fixed up now.

**Review by rhornung67 — APPROVED** [review]
Do calls to the plugin stuff in the code (e.g., in forall, kernel, etc.) need to be guarded?

**Comment by davidbeckingsale** [comment]
> Do calls to the plugin stuff in the code (e.g., in forall, kernel, etc.) need to be guarded?

Nope, that's required for the static stuff - and we decided to leave it in.

**Comment by trws** [comment]
Am I reading this right that effectively the plugin interface and calls will all remain, but effectively the list can't be added to dynamically?

**Comment by davidbeckingsale** [comment]
Good with merging? @rhornung67 @mdavis36

**Comment by rhornung67** [comment]
Yup. Merge away.

**Pull request merged**